src/Symfony/Component/Messenger/Tests/Transport/Doctrine/DoctrineReceiverTest.php

```diff
@@ -77,7 +78,14 @@ public function testOccursRetryableExceptionFromConnection()
         $serializer = $this->createSerializer();
         $connection = $this->createMock(Connection::class);
         $driverException = class_exists(Exception::class) ? Exception::new(new \PDOException('Deadlock', 40001)) : new PDOException(new \PDOException('Deadlock', 40001));
-        $connection->method('get')->willThrowException(new DeadlockException('Deadlock', $driverException));
+        if (!interface_exists(Result::class)) {
```

Comment:
We have previously checked for the presence of the `Doctrine\DBAL\Version` to see if we're on DBAL 3. But I think, we were able to remove that code in the meantime.